akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala

```diff
@@ -21,6 +21,34 @@ import com.github.ghik.silencer.silent
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object ActorContextImpl {
+
+  // single context for logging as there are a few things that are initialized
+  // together
+  final class LoggingContext(val logger: Logger, tags: Set[String]) {
+    // toggled once per message if logging is used to avoid having to
+    // touch the mdc thread local for cleanup in the end
+    var mdcUsed = false
+    val tagsString =
+      // "" means no tags
+      if (tags.isEmpty) ""
```

Comment:
Since we check for this, and ultimately the value is user input, do we also check if null